Add Bitcoin Core config options.

jlopp opened the pull request:
Add all the data mappings for config options and expose them in the UI via the Editor.

Melvillian commented:
bitcoind does not accept boolean config values of "true" or "false" but rather 1 and 0. Replace line 367 of components/Editor.js

`onChange={(ev) => this.change(settings[section], prop)(ev.target.checked ? 1 : 0)}`

so that 0 and 1 will be written to the conf file.

Melvillian commented on src/data.json:
```diff
@@ -11,662 +11,800 @@
+    "walletrbf": {
+      "name": "Enable Replace By Fee Transactions",
+      "description": "Send transactions with full-RBF opt-in enabled.",
+      "default": true
```
default should be `false` since that's the default in 0.14 Core config

Melvillian commented on src/components/Editor.js:
```diff
@@ -24,124 +24,159 @@ class Editor extends Component {
+          { this.flag('debug', 'printtoconsole') }
+          { this.text('debug', 'promiscuousmempoolflags') }
+          { this.flag('debug', 'shrinkdebugfile') }
+          { this.flag('debug', 'testnet') }
```
`testnet` and `regtest` should not be individual boolean options in the `debug` section, but should be a `.select` style option in the `core` section where they user can select [mainnet, testnet3, regtest, simnet] with mainnet (no option) as the default.

jlopp replied:
That would be preferable, though I don't think that the current logic makes it easy to support different variable names for the same selection list.

jlopp commented:
Good catch; I updated that code to output the flag values and 0 and 1 and I updated all the json files to store default values as 0 and 1 rather than booleans.

Melvillian commented on src/components/Presets/Presets.js:
```diff
@@ -25,17 +32,16 @@ function mix (a, b) {
+  'Privacy': tor,
+  'Testnet': testnet,
+  'Regtest': regtest,
+  'Non-syncing': nosync
```
we need to add preset configs for nosync, right now it's empty.

I think all it needs is `connect = 0`

jlopp replied:
Maybe you're looking at an older diff; I added the nosync configs

jlopp commented on src/data.json:
```diff
@@ -11,662 +11,800 @@
+      "default": 20,
+      "min": 1
+    },
+    "datacarrier": {
```
Yeah, they are the same. Though no one ever calls them "data carrier" transactions during normal usage.

jlopp commented:
I'll change the section description lines to start with a '#' so that they are ignored.

You can find most of the config options in 2 places in Core:
https://github.com/bitcoin/bitcoin/blob/master/src/init.cpp#L333
and https://github.com/bitcoin/bitcoin/blob/f34cdcbd806d3e7c9d5dbac2a201755a4cc9828b/src/wallet/wallet.cpp#L3588
